index.js

```diff
@@ -101,44 +101,34 @@ const getAddresses = () => {
 
 
 const hash = (address) => {
-  return Crypto.SHA256(address).toString(Crypto.enc.Hex);
-
+  return Crypto.SHA256(address).toString(Crypto.enc.Hex)
 }
 
 const setTotalActiveUsers = (req) => {
   const address = hash(req.connection.remoteAddress)
-  incrementActiveUsers(address)
-}
-
-incrementActiveUsers = (address) => {
   if(!userActive(address)){ //The last connection was before we zeroed the totalActiveUsers field
     _totalActiveUsers++
   }
 }
 
-decrementActiveUsers = (address) => {
-  if (userActive(address)) { //The last connection was after we zeroed the totalActiveUsers field
-    _totalActiveUsers--
-  }
-}
-
 const userActive = (address) => {  
-  const lastRequestAt = _rlAddressToRequests[address] ? _rlAddressToRequests[address].lastRequestAt : null
+  const lastRequestAt = _rlAddressToRequests[address] && _rlAddressToRequests[address].lastRequestAt
   return (lastRequestAt && lastRequestAt.diff(_lastActiveTimeout) >= 0)
 }
 
 const removeUser = (address) => {
-  decrementActiveUsers(address) //If user is active we must decrement the active users field
+  if (userActive(address)) { //The last connection was after we zeroed the totalActiveUsers field
+    _totalActiveUsers--
+  }
   delete _rlAddressToRequests[address]
```

Comment:
Is there a risk that this will address will be out of range? Or are we sure that this won't throw an error?

Reply:
The removeUser function is called with a shifted value from the _userQueue object. The userQueue and the user table should always be in sync. And if they are, the address will always be in range. However I have no formal proof of this.

Reply:
That's probably enough! Otherwise you could just add `_rlAddressToRequests[address] && delete ...`

Reply:
yes, sure